modules/azurerm_vm/variables.tf

```diff
@@ -1,115 +1,149 @@
 variable "azure_rg_name" {
-  type = string
+  description = "The name of the Resource Group where the VM should exist."
+  type        = string
 }
 
 variable "azure_rg_location" {
-  type = string
+  description = "The name of the Resource Group location where the VM should exist."
+  type        = string
 }
 
 variable "azure_nsg_id" {
-  type = string
+  description = "The ID of the Network Security Group."
+  type        = string
 }
 
 variable "tags" {
   description = "Map of common tags to be placed on the Resources"
-  type        = map
+  type        = map(any)
 }
 
 variable "vnet_subnet_id" {
-  type = string
+  description = "The ID of the Subnet. Changing this forces a new resource to be created."
+  type        = string
 }
 
 variable "machine_type" {
-  default = "Standard_E8s_v3"
+  description = "The size which should be used for this Virtual Machine, such as Standard_F2."
+  type        = string
+  default     = "Standard_E8s_v3"
 }
 
 variable "vm_admin" {
   description = "OS Admin User for VMs of AKS Cluster nodes"
+  type        = string
   default     = "azureuser"
 }
 
 variable "vm_zone" {
   description = "The Zone in which this Virtual Machine should be created. Changing this forces a new resource to be created"
+  type        = string
   default     = null
 }
 
 variable "ssh_public_key" {
   description = "Path to ssh public key"
+  type        = string
   default     = ""
 }
 
 variable "os_publisher" {
-  default = "Canonical"
+  description = "Specifies the publisher of the image used to create the virtual machines. Changing this forces a new resource to be created."
+  type        = string
+  default     = "Canonical"
 }
 
 variable "os_offer" {
-  default = "0001-com-ubuntu-server-focal"
+  description = "Specifies the offer of the image used to create the virtual machines. Changing this forces a new resource to be created."
+  type        = string
+  default     = "0001-com-ubuntu-server-focal"
 }
 
 variable "os_sku" {
-  default = "20_04-lts"
+  description = "Specifies the SKU of the image used to create the virtual machines. Changing this forces a new resource to be created."
+  type        = string
+  default     = "20_04-lts"
 }
 
 variable "os_version" {
-  default = "latest"
+  description = "Specifies the version of the image used to create the virtual machines. Changing this forces a new resource to be created."
+  type        = string
+  default     = "latest"
 }
 
-variable name {
-  type = string
+variable "name" {
+  description = "VM name"
+  type        = string
 }
 
-variable data_disk_count {
-  default = 0
+variable "data_disk_count" {
+  description = "Number of disks to be created"
+  type        = number
+  default     = 0
 }
 
-variable data_disk_size {
-  default = 128
+variable "data_disk_size" {
+  description = "Specifies the size of the managed disk to create in gigabytes."
```

Comment:
noticed that GB was used in other locations but spelled out here, if it makes sense could use the same for all.

Reply:
Updated the description in the latest commit.